### PR TITLE
Schema Design

### PR DESCRIPTION
**Pull request by ericharmeling**
Really only the last commit needs reviewing:

https://github.com/cockroachdb/docs/commit/740f4f3fb126ed7bd4d398f87064ebd798cf15a4

The following pages in particular:
- `v21.1/schema-design-database.md` (second review)
- `v21.1/schema-design-schema.md` (second review)
- `v21.1/schema-design-table.md` (second review)
- `v21.1/schema-design-index.md` (first full review)
- `v21.1/schema-design-update.md` (first full review)

Thanks!

### CODE REVIEW

File: v21.1/schema-design-update.md [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+~~~
+
+Under that first `ALTER TABLE` statement, add another `ALTER TABLE` statement for changing the primary key columns to `username` and `email`:
+
+{% include copy-clipboard.html %}
+~~~
+ALTER TABLE IF EXISTS movr.max_schema.users ALTER PRIMARY KEY USING COLUMNS (username, email);
+~~~
+
+In order to add a column to an existing table's primary key index, the column must have an existing [`NOT NULL` constraint](not-null.html). Neither the `username` nor the `email` tables have `NOT NULL` constraints.
```
Comment (ianjevans) [inline v21.1/schema-design-update.md:110]:
```suggestion
In order to add a column to an existing table's primary key index, the column must have an existing [`NOT NULL` constraint](not-null.html). Neither the `username` nor the `email` columns have `NOT NULL` constraints.
```

**Review by ianjevans — COMMENTED** [review]
Just some minor changes.

File: v21.1/schema-design-update.md (hunk 1 above)
Comment (ericharmeling) [reply v21.1/schema-design-update.md]: Done.

File: v21.1/schema-design-schema.md [hunk 2]
```diff
@@ -39,115 +39,130 @@ Before reading this page, do the following:
...
+The first statement sets the `movr` database as the [current database](sql-name-resolution.html#current-database). The next two sets of statements create SQL users named `max` and `abbey` in the `movr` database, with [`CREATE` privileges on the database](authorization.html#supported-privileges). `CREATE` privileges will allow each user to create tables in the database.
 
-Open `max_dbinit.sql` in a text editor of your choice, and add a `CREATE SCHEMA` statement to the top of the file:
+Now, under the `CREATE USER` statements, add `DROP SCHEMA` and `CREATE SCHEMA` statements for each user's user-defined schema:
 
 {% include copy-clipboard.html %}
 ~~~
-CREATE SCHEMA IF NOT EXISTS max_schema;
+DROP SCHEMA IF EXISTS max_schema CASCADE;
+CREATE SCHEMA IF NOT EXISTS max_schema AUTHORIZATION max;
```
[the remark this answers is not in the document]
Comment (ericharmeling) [reply v21.1/schema-design-schema.md]: Good point. Removed the `IF NOT EXISTS` clause.

File: v21.1/schema-design-indexes.md [hunk 3]
```diff
@@ -4,4 +4,280 @@ summary: Index columns in your schema
...
+
+    Also note that [`ALTER PRIMARY KEY`](alter-primary-key.html) creates a secondary index from the old primary key. If you need to [change a primary key](constraints.html#change-constraints), and you do not plan to filter queries on the old primary key column(s), do not use `ALTER PRIMARY KEY`. Instead, use [`DROP CONSTRAINT ... PRIMARY KEY`/`ADD CONSTRAINT ... PRIMARY KEY`](add-constraint.html#changing-primary-keys-with-add-constraint-primary-key), which does not create a secondary index.
+
+- Use a [`STORING` clause](create-index.html#parameters) to store columns of data that you want returned by common queries, but that you do not plan to use in query filters.
+
+    The `STORING` clause specifies columns that are not part of the index key but should be stored in the index, without being sorted. If a column is specified in a query, and the column is neither indexed nor stored in an index, CockroachDB will perform a full scan of the table, which can have a significant impact on performance. For an example, see [below](#examples).
```
[the remark this answers is not in the document]
Comment (ericharmeling) [reply v21.1/schema-design-indexes.md]: Done.

File: v21.1/schema-design-indexes.md [hunk 4]
```diff
@@ -4,4 +4,280 @@ summary: Index columns in your schema
...
+
+    ~~~
+    INDEX [index_name] ([column_names]);
+    ~~~
+
+    Where `[index_name]` is the name of the index, and `[column_names]` is the name of the column to index, or a comma-separated list of names of the columns to index.
+
+- Use a [`CREATE INDEX`](create-index.html) statement.
+
+    `CREATE INEX` statements generally take the following form:
```
[the remark this answers is not in the document]
Comment (ericharmeling) [reply v21.1/schema-design-indexes.md]: Done.

File: _includes/v21.1/sql/dev-schema-change-limits.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Review the [limitations of online schema changes in CockroachDB](online-schema-changes.html#limitations). In specific, note that CockroachDB has [limited support for schema changes within the same explicit transaction](online-schema-changes#limited-support-for-schema-changes-within-transactions).
+
+    We recommend doing schema changes outside explicit transactions, where possible. When a database schema management tool manages transactions on your behalf, we recommend only including one schema change operation per transaction.
```
[the remark this answers is not in the document]
Comment (ericharmeling) [reply _includes/v21.1/sql/dev-schema-change-limits.md]: Done.

File: _includes/v21.1/sql/dev-schema-change-limits.md [hunk 6]
```diff
@@ -0,0 +1,3 @@
+Review the [limitations of online schema changes in CockroachDB](online-schema-changes.html#limitations). In specific, note that CockroachDB has [limited support for schema changes within the same explicit transaction](online-schema-changes#limited-support-for-schema-changes-within-transactions).
```
[the remark this answers is not in the document]
Comment (ericharmeling) [reply _includes/v21.1/sql/dev-schema-change-limits.md]: Done.

**Pull request merged**